mm2src/coins/nft/storage/wasm/wasm_storage.rs

```diff
@@ -28,144 +31,529 @@ impl IndexedDbNftStorage {
+        let last_scanned_block_table = db_transaction.table::<LastScannedBlockTable>().await?;
+        for nft in nfts {
+            let nft_item = NftListTable::from_nft(&nft)?;
+            nft_table.add_item(&nft_item).await?;
+        }
+        let last_scanned_block = LastScannedBlockTable {
+            chain: chain.to_string(),
+            last_scanned_block: BeBigUint::from(last_scanned_block),
+        };
+        last_scanned_block_table
+            .replace_item_by_unique_index("chain", chain.to_string(), &last_scanned_block)
+            .await?;
+        Ok(())
     }
 
     async fn get_nft(
         &self,
-        _chain: &Chain,
-        _token_address: String,
-        _token_id: BigDecimal,
+        chain: &Chain,
+        token_address: String,
+        token_id: BigDecimal,
     ) -> MmResult<Option<Nft>, Self::Error> {
-        todo!()
+        let locked_db = self.lock_db().await?;
+        let db_transaction = locked_db.get_inner().transaction().await?;
+        let table = db_transaction.table::<NftListTable>().await?;
+        let index_keys = MultiIndex::new(NftListTable::CHAIN_TOKEN_ADD_TOKEN_ID_INDEX)
+            .with_value(chain.to_string())?
+            .with_value(&token_address)?
+            .with_value(token_id.to_string())?;
+
+        if let Some((_item_id, item)) = table.get_item_by_unique_multi_index(index_keys).await? {
+            Ok(Some(nft_details_from_item(item)?))
+        } else {
+            return Ok(None);
+        }
     }
 
     async fn remove_nft_from_list(
         &self,
-        _chain: &Chain,
-        _token_address: String,
-        _token_id: BigDecimal,
-        _scanned_block: u64,
+        chain: &Chain,
+        token_address: String,
+        token_id: BigDecimal,
+        scanned_block: u64,
     ) -> MmResult<RemoveNftResult, Self::Error> {
-        todo!()
+        let locked_db = self.lock_db().await?;
+        let db_transaction = locked_db.get_inner().transaction().await?;
+        let nft_table = db_transaction.table::<NftListTable>().await?;
+        let last_scanned_block_table = db_transaction.table::<LastScannedBlockTable>().await?;
+
+        let index_keys = MultiIndex::new(NftListTable::CHAIN_TOKEN_ADD_TOKEN_ID_INDEX)
+            .with_value(chain.to_string())?
+            .with_value(&token_address)?
+            .with_value(token_id.to_string())?;
+
+        let last_scanned_block = LastScannedBlockTable {
+            chain: chain.to_string(),
+            last_scanned_block: BeBigUint::from(scanned_block),
+        };
+
+        let nft_removed = nft_table.delete_item_by_unique_multi_index(index_keys).await?.is_some();
+        last_scanned_block_table
+            .replace_item_by_unique_index("chain", chain.to_string(), &last_scanned_block)
+            .await?;
+        if nft_removed {
+            Ok(RemoveNftResult::NftRemoved)
+        } else {
+            Ok(RemoveNftResult::NftDidNotExist)
+        }
```

Comment:
Q: if `nft_removed` was false, should we directly return `Ok(RemoveNftResult::NftDidNotExist)` without running `replace_item_by_unique_index` ?

Reply:
We need to update scanned blocks. I also was asking this question myself.
Is it possible that the nft has been deleted but the block has not been updated?
If nft was deleted and block was updated then nothing will be changed, if nft was deleted, but block wasnt scanned, then it should be updated.

Reply:
> Is it possible that the nft has been deleted but the block has not been updated?

How do we know `nft was deleted`, isn't that through the blocks?

Reply:
> How do we know `nft was deleted`, isn't that through the blocks?
we iterate trough transactions to update nft list, if we dont update last scanned block during iteration we will have problems.